clients/da-vinci-client/src/main/java/com/linkedin/davinci/consumer/BootstrappingVeniceChangelogConsumerDaVinciRecordTransformerImpl.java

```diff
@@ -0,0 +1,333 @@
+    }
+
+    return this.start(allPartitions);
+  }
+
+  @Override
+  public void stop() throws Exception {
+    daVinciClientFactory.close();
+    isStarted = false;
+  }
+
+  @Override
+  public Collection<PubSubMessage<K, ChangeEvent<V>, VeniceChangeCoordinate>> poll(long timeoutInMs) {
+    try {
+      Thread.sleep(timeoutInMs);
+    } catch (InterruptedException e) {
+      LOGGER.info("Thread was interrupted", e);
+      // Restore the interrupt status
+      Thread.currentThread().interrupt();
+    }
+
+    List<PubSubMessage<K, ChangeEvent<V>, VeniceChangeCoordinate>> drainedPubSubMessages = new ArrayList<>();
+    pubSubMessages.drainTo(drainedPubSubMessages);
+
+    if (changelogClientConfig.shouldCompactMessages()) {
+      Map<K, PubSubMessage<K, ChangeEvent<V>, VeniceChangeCoordinate>> tempMap = new LinkedHashMap<>();
+      /*
+       * The behavior of LinkedHashMap is such that it maintains the order of insertion, but for values which are
+       * replaced, it's put in at the position of the first insertion. This isn't quite what we want, we want to keep
+       * only a single key (just as a map would), but we want to keep the position of the last insertion as well. So in
+       * order to do that, we remove the entry before inserting it.
+       */
+      for (PubSubMessage<K, ChangeEvent<V>, VeniceChangeCoordinate> message: drainedPubSubMessages) {
+        tempMap.remove(message.getKey());
+        tempMap.put(message.getKey(), message);
+      }
+      return tempMap.values();
+    }
+    return drainedPubSubMessages;
+  }
+
+  private void internalStart() {
+    if (isStarted) {
+      return;
+    }
+
+    daVinciClient.start();
+    isStarted = true;
+  }
+
+  private VeniceProperties buildVeniceConfig() {
+    return new PropertyBuilder().put(ROCKSDB_LEVEL0_FILE_NUM_COMPACTION_TRIGGER, 4) // RocksDB
+        // default config
+        .put(ROCKSDB_LEVEL0_SLOWDOWN_WRITES_TRIGGER, 20) // RocksDB default config
+        .put(ROCKSDB_LEVEL0_STOPS_WRITES_TRIGGER, 36) // RocksDB default config
+        .put(ROCKSDB_LEVEL0_FILE_NUM_COMPACTION_TRIGGER_WRITE_ONLY_VERSION, 40)
+        .put(ROCKSDB_LEVEL0_SLOWDOWN_WRITES_TRIGGER_WRITE_ONLY_VERSION, 60)
+        .put(ROCKSDB_LEVEL0_STOPS_WRITES_TRIGGER_WRITE_ONLY_VERSION, 80)
+        .put(ROCKSDB_BLOCK_CACHE_SIZE_IN_BYTES, changelogClientConfig.getRocksDBBlockCacheSizeInBytes())
+        .put(changelogClientConfig.getConsumerProperties())
+        .put(DATA_BASE_PATH, changelogClientConfig.getBootstrapFileSystemPath())
+        .put(ROCKSDB_PLAIN_TABLE_FORMAT_ENABLED, false)
+        .put(PERSISTENCE_TYPE, ROCKS_DB)
+        // Turning this off, so users don't subscribe to unwanted partitions automatically
+        .put(DA_VINCI_SUBSCRIBE_ON_DISK_PARTITIONS_AUTOMATICALLY, false)
+        .put(BLOB_TRANSFER_MANAGER_ENABLED, changelogClientConfig.isBlobTransferEnabled())
+        .put(CLIENT_USE_SYSTEM_STORE_REPOSITORY, true)
+        .put(PUSH_STATUS_STORE_ENABLED, true)
```

Comment:
I 'think' we should also set PARTICIPANT_MESSAGE_STORE_ENABLED to false.  Since this CDC user is going to be throttling ingestion based on the user calls to poll, it may block version pushes as it's not ready to serve based on customer code.